### PR TITLE
USHIFT-4385: Add devenv instructions on mounting shared directory

### PR DESCRIPTION
**Pull request by ggiguash**
This change also contains an optimization for how hadolint is used. This is necessary to avoid SELinux context setting on an NFS-mounted partition. 

Note that `container_file_t` SELinux context is still required for running container images off the mounted directory. But, unlike the context for hadolint that was always set on verify, the `container_file_t` is only set in case of NFS mount.

### CODE REVIEW

File: docs/contributor/devenv_setup.md [hunk 1]
```diff
@@ -307,3 +307,108 @@ sudo lvextend -l +95%FREE /dev/mapper/rhel-root
...
+One way of using the shared directory would be to keep the whole source tree in
+it for easy access from the host.
+
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo mkdir -p -m 0755 "${MOUNT_DIR}/microshift"
+sudo chown $(whoami). "${MOUNT_DIR}/microshift"
+
+ln -s "${MOUNT_DIR}/microshift" ~/microshift
```
Comment (pmtk) [inline docs/contributor/devenv_setup.md:413]: I think this misses git clone? Or at least move of the repo?
I already have source in ~/microshift, so the last command creates a symlink at `/home/microshift/microshift/microshift` pointing to `/home/microshift-shared/microshift` so it's invalid.

Also is it a bit weird that dev vm populates host's share? Why not share the microshift repo from the host directly mounted as ~/microshift in the VM?

File: docs/contributor/devenv_setup.md [hunk 2]
```diff
@@ -307,3 +307,108 @@ sudo lvextend -l +95%FREE /dev/mapper/rhel-root
...
+```
+
+#### Synchronize User and Group Identifiers
+
+For read-write to work properly both on the hypervisor host and on the virtual
+machine, it is recommended to edit the `microshift` user and group
+identifiers on the virtual machine to have the value as on the hypervisor.
+
+Edit the `/etc/passwd` and `/etc/group` files on the virtual machine to replace
+the `microshift` user and group identifiers.
```
[the remark this answers is not in the document]
Comment (ggiguash) [reply docs/contributor/devenv_setup.md:385]: Added more explanation.

File: docs/contributor/devenv_setup.md (hunk 1 above)
Comment (ggiguash) [reply docs/contributor/devenv_setup.md:413]: Having the whole home directory remotely mounted maybe a bit too intrusive. I have not tested it, but I can add a note that this is another sharing option.

File: docs/contributor/devenv_setup.md [hunk 3]
```diff
@@ -307,3 +307,137 @@ sudo lvextend -l +95%FREE /dev/mapper/rhel-root
...
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo mkdir -p -m 0755 "${MOUNT_DIR}/microshift"
+sudo chown $(whoami). "${MOUNT_DIR}/microshift"
+
+# Move the existing home directory
+[ -d /home/microshift ] && sudo mv /home/microshift /home/microshift.local
+# Do not attempt to create the link if the target already exists
+[ ! -L /home/microshift ] && sudo ln -s "${MOUNT_DIR}" /home/microshift
```
Comment (pmtk) [inline docs/contributor/devenv_setup.md:442]: I think this goes too far. I don't think it's a good advice to move existing home directory and mount whole share as home dir. What about all the dotfiles there, like ~/.config, ~/.ssh, ~/.local/bin, and others?

**Pull request merged**